uspace/lib/c/generic/async/server.c

```diff
@@ -190,6 +194,15 @@ typedef struct {
 
 	/** Notification data */
 	void *data;
+
+	/**
+	 * Data of the most recent notification.
+	 * If count > 1, the data for previous notifications are lost.
+	 */
+	ipc_call_t calldata;
+
+	/** How many of this notification arrived since it was last handled. */
+	long count;
```

Comment:
Should this be rather unsigned?

Reply:
I have a rule not to use unsigned unless either the value is a bitfield, or I want wrapping overflows. With signed types, you can make unexpected overflows fatal using `-ftrapv`, or detect them with undefined behavior sanitizer. No such escape hatch exists for unsigned, since for them overflows are legal.